core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/dequeue_coordinator.go

```diff
@@ -0,0 +1,126 @@
+package logprovider
+
+import (
+	"sync"
+)
+
+type DequeueCoordinator interface {
```

Comment:
we have already coordinator in a different context, and it seems that the interface exposes something more like `LogBufferIndex` that stores metadata and desired information of the buffer in structured collections.